magicblock-committor-service/src/intent_executor/mod.rs

```diff
@@ -415,6 +573,193 @@ where
+        &self,
+        prepared_message: VersionedMessage,
+        tasks: &[Box<dyn BaseTask>],
+    ) -> IntentExecutorResult<Signature, TransactionStrategyExecutionError>
+    {
+        const RETRY_FOR: Duration = Duration::from_secs(2 * 60);
+        const MIN_RETRIES: usize = 3;
+
+        const SLEEP: Duration = Duration::from_millis(500);
+
+        let decide_flow_rpc_error =
+            |err: solana_rpc_client_api::client_error::Error,
+             map: fn(
+                solana_rpc_client_api::client_error::Error,
+            ) -> MagicBlockRpcClientError|
+             -> ControlFlow<
+                TransactionStrategyExecutionError,
+                TransactionStrategyExecutionError,
+            > {
+                let map_helper =
+                    |request, kind| -> TransactionStrategyExecutionError {
+                        TransactionStrategyExecutionError::InternalError(
+                            InternalError::MagicBlockRpcClientError(map(
+                                solana_rpc_client_api::client_error::Error {
+                                    request,
+                                    kind,
+                                },
+                            )),
+                        )
+                    };
+
+                match err.kind {
+                    ErrorKind::TransactionError(transaction_err) => {
+                        // Map transaction error to a known set, otherwise maps to internal error
+                        // We're returning immediately to recover
+                        let error = TransactionStrategyExecutionError::from_transaction_error(transaction_err, tasks, |transaction_err| -> MagicBlockRpcClientError {
+                            map(
+                                solana_rpc_client_api::client_error::Error {
+                                    request: err.request,
+                                    kind: ErrorKind::TransactionError(transaction_err),
+                                })
+                        });
+
+                        ControlFlow::Break(error)
+                    }
+                    err_kind @ ErrorKind::Io(_) => {
+                        // Attempting retry
+                        ControlFlow::Continue(map_helper(err.request, err_kind))
+                    }
+                    err_kind @ (ErrorKind::Reqwest(_)
+                    | ErrorKind::Middleware(_)
+                    | ErrorKind::RpcError(_)
+                    | ErrorKind::SerdeJson(_)
+                    | ErrorKind::SigningError(_)
+                    | ErrorKind::Custom(_)) => {
+                        // Can't handle - propagate
+                        ControlFlow::Break(map_helper(err.request, err_kind))
+                    }
+                }
+            };
+
+        // Initialize with a default error to avoid uninitialized variable issues
+        let mut last_err = TransactionStrategyExecutionError::InternalError(
```

Comment:
This _hack_ is just another indicator that this `last_err` approach is not ideal.

Why can't we collect errors we see in a vec and include them all or combine them into an aggregated error later?